File: IkiBoyutluMatrisAnalizi/IkiBoyutluMatrisAnalizi.java
package ikiboyutlumatrisanalizi;

public class IkiBoyutluMatrisAnalizi {

    public static void main(String[] args) {
        int[][] matris = new int[10][10];
        for (int i = 0; i < matris.length; i++) {
            for (int j = 0; j < matris[0].length; j++) {
                int rastgele = (int) (Math.random() * 10);
                matris[i][j] = rastgele;
                System.out.print(matris[i][j] + "\t");
            }
            System.out.println("");
        }
        int sagdanCaprazToplam = 0, soldanCaprazToplam = 0;
        for (int i = 0; i < matris.length; i++) {
            for (int j = 0; j < matris[0].length; j++) {
                if (i == j) {
                    soldanCaprazToplam += matris[i][j];
                }
                if (i + j == matris.length - 1) {
                    sagdanCaprazToplam += matris[i][j];
                }
            }
        }
        System.out.println("Soldan çarpaz toplam : " + soldanCaprazToplam + "\nSağdan çapraz toplam : " + sagdanCaprazToplam);
    }
}

File: IkiBoyutluMatrisAnalizi/IkiBoyutluMatrisAnalizi.py
import random

def matris_analizi():

    matris = [[0 for x in range(10)]for y in range(10)]
    sagdan_capraz_toplam = 0
    soldan_capraz_toplam = 0

    for i in range(10):
        for j in range(10):
            matris[i][j] = random.randint(1, 9)
            print(matris[i][j], end='\t')
        print()

    for i in range(10):
        for j in range(10):
            if i == j:
                soldan_capraz_toplam += matris[i][j]
            if i + j == len(matris[0])-1:
                sagdan_capraz_toplam += matris[i][j]

    print("Soldan çarpaz toplam : {}\nSağdan çapraz toplam : {}".format(soldan_capraz_toplam, sagdan_capraz_toplam))

matris_analizi()